packages/firebase_messaging/android/src/main/java/io/flutter/plugins/firebasemessaging/FirebaseMessagingPlugin.java

```diff
@@ -99,7 +110,28 @@ public void onReceive(Context context, Intent intent) {
 
   @Override
   public void onMethodCall(final MethodCall call, final Result result) {
-    if ("configure".equals(call.method)) {
+    if ("FcmDartService.start".equals(call.method)) {
```

Comment:
I think some thorough comments in this method might be helpful. Very few developers (including framework developers) are familiar with Dart callback registration and invocation, as well as Android Service behaviors. I would recommend err'ing on the side of an abundance of comments that clarify these details...

Reply:
Added comments describing the need for `FlutterFirebaseMessagingService` and the methods `FcmDartService#start` and `FcmDartService#initialized`